### PR TITLE
Support track creation/retrieval

### PR DESCRIPTION
**Pull request by viethang**

### CODE REVIEW

File: api/tracks/views.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  def get_queryset(self):
+    user_id = self.kwargs.get('user_id')
+    return models.Track.objects.filter(user=user_id)
+
+  def post(self, request, user_id):
+    request.data['user'] = user_id
+    serializer = TrackSerializer(data=request.data)
+    if serializer.is_valid():
+      serializer.save()
+      return Response(serializer.data, status=status.HTTP_201_CREATED)
```
Comment (gberaudo) [inline api/tracks/views.py]: Do passing a track with an id hijack the original track? If yes I guess it should also hijack tracks from other users.

File: api/tracks/views.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+class UserOneTrackView(RetrieveUpdateDestroyAPIView):
+  serializer_class = TrackSerializer
+  permission_classes = (SameUserOrAdminPermission, IsAdminOrTrackOwner)
+
+  def get(self, request, user_id, track_id):
+    try:
+      track = models.Track.objects.get(id=track_id)
+      return Response(TrackSerializer(track).data)
+    except models.Track.DoesNotExist:
+      return Response(status=status.HTTP_400_BAD_REQUEST)
```
Comment (gberaudo) [inline api/tracks/views.py]: Shouldn't it be a 404 error?

File: api/tracks/views.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+
+  def put(self, request, user_id, track_id):
+    try:
+      track = models.Track.objects.get(id=track_id)
+      serializer = TrackSerializer(track)
+      serializer.update(track, request.data)
```
Comment (gberaudo) [inline api/tracks/views.py]: What happens if we pass bad data like non-existing fields?

File: api/tracks/views.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      return Response(serializer.data)
+    except models.Track.DoesNotExist:
+      return Response(status=status.HTTP_400_BAD_REQUEST)
+
+  def destroy(self, request, user_id, track_id):
+    try:
+      track = models.Track.objects.get(id=track_id)
+      track.delete()
```
Comment (gberaudo) [inline api/tracks/views.py]: It should be possible to delete the track without loading it with something like:
`models.Track.filter(id=track_id).delete()`.

File: api/tracks/views.py (hunk 1 above)
Comment (viethang) [reply api/tracks/views.py]: No hijack possible here, according to Django Rest's doc
https://www.django-rest-framework.org/api-guide/serializers/#saving-instances

# .save() will create a new instance.
serializer = CommentSerializer(data=data)

# .save() will update the existing `comment` instance.
serializer = CommentSerializer(comment, data=data)

File: api/tracks/views.py (hunk 3 above)
Comment (viethang) [reply api/tracks/views.py]: It just ignores fields that are not define in TrackSerializer.Meta.fields

File: api/tracks/views.py (hunk 2 above)
Comment (viethang) [reply api/tracks/views.py]: OK

File: api/tracks/views.py (hunk 4 above)
Comment (viethang) [reply api/tracks/views.py]: Ok, good idea.

**Pull request merged**